pkg/synchronization/controller.go

```diff
@@ -534,6 +539,41 @@ func (c *Controller) createOrUpdateSecret(logger *zap.Logger, secretSpec *core_v
 	return true, secret, err
 }
 
+func (c *Controller) ensureCommonSecretExists(logger *zap.Logger, ns *core_v1.Namespace) (bool /* retriable */, error) {
```

Comment:
Probably doesn't need to whole `ns` object, but I guess you are just following the style used in other functions. np